.github/workflows/deploy-personal-dev-environment.yml

```diff
@@ -63,30 +69,126 @@ jobs:
 
       - name: Build frontend image
         run: |
+          cd data-dashboard-infra
           source uhd.sh
-          uhd gh co frontend {{ inputs.frontend_branch }}
+          uhd terraform output ${{ inputs.name }}
           uhd docker build frontend
         shell: zsh {0}
 
+  push_backend_docker_image:
+    name: Push backend docker image
+    runs-on: ubuntu-latest
+    needs: [ "terraform_apply" ]
+    steps:
+      - name: Checkout .github/ directory
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
+
+      - name: Configure AWS credentials for tools account
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          tools-account-role: ${{ secrets.UHD_TERRAFORM_IAM_ROLE }}
+
+      - uses: ./.github/actions/setup-zsh
+
+      - name: Checkout backend repo
+        uses: actions/checkout@v4
+        with:
+          repository: UKHSA-Internal/data-dashboard-api
+          path: data-dashboard-api
+          ref: ${{ inputs.backend_branch }}
+
+      - name: Checkout infra repo
+        uses: actions/checkout@v4
+        with:
+          path: data-dashboard-infra
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
       - name: Build backend image
         run: |
+          cd data-dashboard-infra
           source uhd.sh
-          uhd gh co api {{ inputs.backend_branch }}
-          uhd docker build backend
+          uhd terraform output ${{ inputs.name }}
+          uhd docker build ingestion
+        shell: zsh {0}
+
+  push_ingestion_docker_image:
+    name: Push ingestion docker image
+    runs-on: ubuntu-latest
+    needs: [ "terraform_apply" ]
+    steps:
+      - name: Checkout .github/ directory
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
+
+      - name: Configure AWS credentials for tools account
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          tools-account-role: ${{ secrets.UHD_TERRAFORM_IAM_ROLE }}
+
+      - uses: ./.github/actions/setup-zsh
+
+      - name: Checkout backend repo
+        uses: actions/checkout@v4
+        with:
+          repository: UKHSA-Internal/data-dashboard-api
+          path: data-dashboard-api
+          ref: ${{ inputs.backend_branch }}
+
+      - name: Checkout infra repo
+        uses: actions/checkout@v4
+        with:
+          path: data-dashboard-infra
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build ingestion image
+        run: |
+          cd data-dashboard-infra
+          source uhd.sh
+          uhd terraform output ${{ inputs.name }}
+          uhd docker build ingestion
```

Comment:
This is all quite messy and repetitve at the moment.
But I just wanna get it in so people can start using it, I'm planning on coming back and DRYing out this into a  reuseable action